app/src/main/kotlin/batect/cli/options/OptionParser.kt

```diff
@@ -41,17 +41,11 @@ class OptionParser {
     private fun parseOption(args: Iterable<String>, currentIndex: Int): OptionParsingResult {
         val arg = args.elementAt(currentIndex)
         val argName = arg.substringBefore("=")
-        val option = optionNames[argName]
-
-        if (option == null) {
-            if (argName.startsWith("-")) {
-                return OptionParsingResult.InvalidOption("Invalid option '$argName'. Run './batect --help' for a list of valid options.")
-            } else {
-                return OptionParsingResult.NoOption
-            }
+        return optionNames[argName]?.parse(args.drop(currentIndex)) ?: if (argName.startsWith("-")) {
+            OptionParsingResult.InvalidOption("Invalid option '$argName'. Run './batect --help' for a list of valid options.")
+        } else {
+            OptionParsingResult.NoOption
         }
-
-        return option.parse(args.drop(currentIndex))
```

Comment:
Sorry, this is far less clear to me. Please revert this change.

Reply:
I've mostly reverted the change, but still applied the `'Return' can be lifted out of 'if'` suggestion. I've *not* applied the also present `If-Null return/break/... foldable to '?:'` suggestion, as I believe that's the bit which makes it less readable for you. Feel free to play around with it, though, as there might be a compromise, like

        val option = optionNames[argName]
            ?: return if (argName.startsWith("-")) {
                OptionParsingResult.InvalidOption("Invalid option '$argName'. Run './batect --help' for a list of valid options.")
            } else {
                OptionParsingResult.NoOption
            }

        return option.parse(args.drop(currentIndex))